samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c

```diff
@@ -0,0 +1,562 @@
+	mqtt_client_init(client);
+	client->broker = &broker;
+	client->evt_cb = mqtt_event_handler;
+	client->client_id.utf8 = client_id;
+	client->client_id.size = strlen(client->client_id.utf8);
+	client->password = NULL;
+	client->user_name = NULL;
+	client->protocol_version = MQTT_VERSION_3_1_1;
+
+	/* MQTT buffers configuration */
+	client->rx_buf = rx_buffer;
+	client->rx_buf_size = sizeof(rx_buffer);
+	client->tx_buf = tx_buffer;
+	client->tx_buf_size = sizeof(tx_buffer);
+
+	/* MQTT transport configuration */
+#if defined(CONFIG_MQTT_LIB_TLS)
+	client->transport.type = MQTT_TRANSPORT_SECURE;
+
+	if (tls_init() != 0) {
+		LOG_ERR("TLS init error");
+		return;
+	}
+
+	struct mqtt_sec_config *tls_config;
+
+	tls_config = &client->transport.tls.config;
+	tls_config->peer_verify = TLS_PEER_VERIFY_REQUIRED;
+	tls_config->cipher_list = NULL;
+	tls_config->sec_tag_list = m_sec_tags;
+	tls_config->sec_tag_count = ARRAY_SIZE(m_sec_tags);
+	#if defined(CONFIG_MBEDTLS_SERVER_NAME_INDICATION)
+	tls_config->hostname = CONFIG_MQTT_BROKER_HOSTNAME;
+	#else
+	tls_config->hostname = NULL;
+	#endif /* CONFIG_MBEDTLS_SERVER_NAME_INDICATION */
+#endif
+}
+
+static int poll_mqtt_socket(struct mqtt_client *client, int timeout)
+{
+	int rc;
+
+	prepare_fds(client);
+
+	if (nfds <= 0) {
+		return -EINVAL;
+	}
+
+	rc = zsock_poll(fds, nfds, timeout);
+	if (rc < 0) {
+		LOG_ERR("Socket poll error [%d]", rc);
+	}
+
+	return rc;
+}
+
+/** Retrieves a sensor sample and encodes it in JSON format */
+static int get_mqtt_payload(struct mqtt_binstr *payload)
+{
+	int rc;
+	struct sensor_sample sample;
+
+	rc = device_read_sensor(&sample);
+	if (rc != 0) {
+		LOG_ERR("Failed to get sensor sample [%d]", rc);
+		return rc;
+	}
+
+	rc = json_obj_encode_buf(sensor_sample_descr, ARRAY_SIZE(sensor_sample_descr),
+					&sample, payload_buf, CONFIG_MQTT_PAYLOAD_SIZE);
+	if (rc != 0) {
+		LOG_ERR("Failed to encode JSON object [%d]", rc);
+		return rc;
+	}
+
+	payload->data = payload_buf;
+	payload->len = strlen(payload->data);
+
+	return rc;
+}
+
+/** Perform publish of MQTT payload */
+int app_mqtt_publish(struct mqtt_client *client)
+{
+	int rc;
+	struct mqtt_publish_param param;
+	struct mqtt_binstr payload;
+	static uint16_t msg_id = 1;
+	struct mqtt_topic topic = {
+		.topic = {
+			.utf8 = CONFIG_MQTT_PUB_TOPIC,
+			.size = strlen(topic.topic.utf8)
+		},
+		.qos = CONFIG_MQTT_PUBLISH_QOS
+	};
+
+	rc = get_mqtt_payload(&payload);
+	if (rc != 0) {
+		LOG_ERR("Failed to get MQTT payload [%d]", rc);
+	}
+
+	param.message.topic = topic;
+	param.message.payload = payload;
+	param.message_id = msg_id++;
+	param.dup_flag = 0;
+	param.retain_flag = 0;
+
+	rc = mqtt_publish(client, &param);
+	if (rc != 0) {
+		LOG_ERR("MQTT Publish failed [%d]", rc);
+	}
+
+	LOG_INF("Published to topic '%s', QoS %d",
+			param.message.topic.topic.utf8,
+			param.message.topic.qos);
+
+	return rc;
+}
+
+/** Subscribe to defined MQTT topics */
+int app_mqtt_subscribe(struct mqtt_client *client)
+{
+	int rc;
+
```

Comment:
No need for empty lines around variable definitions.

Reply:
Done.